app/Console/Commands/ImportNasspGroupsCommand.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace Rogue\Console\Commands;
+
+use Exception;
+use League\Csv\Reader;
+use Rogue\Models\Group;
+use Rogue\Models\GroupType;
+use Illuminate\Console\Command;
+
+class ImportNasspGroupsCommand extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'rogue:nassp-groups-import {path}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Create National Association of Secondary School Principals group types and groups from a CSV.';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Returns a trimmed CSV value, or null if equal to the string "NULL".
+     *
+     * @param string
+     * @return string
+     */
+    public function sanitize($value)
+    {
+        $result = trim($value);
+
+        return $result == 'NULL' ? null : $value;
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        // Sample CSV path: https://gist.githubusercontent.com/aaronschachter/a407e9882ebe01520644cc9c257c8ec9/raw/8c837c52cfe22866ba4b49f2cf2f12336ca297d8/groups.csv
+
+        $path = $this->argument('path');
+
+        info('rogue:nassp-groups-import: Loading csv from ' . $path);
+
+        // Make a local copy of the CSV.
+        $temp = tempnam(sys_get_temp_dir(), 'command_csv');
+
+        file_put_contents($temp, fopen($this->argument('path'), 'r'));
+
+        // Load CSV contents.
+        $csv = Reader::createFromPath($temp, 'r');
+
+        $csv->setHeaderOffset(0);
+
+        $numImported = 0;
+        $numFailed = 0;
+        $numSkipped = 0;
+
+        $nhsGroupType = GroupType::firstOrCreate([
+            'name' => 'National Honor Society',
+        ]);
+        $njhsGroupType = GroupType::firstOrCreate([
+            'name' => 'National Junior Honor Society',
+        ]);
+        $nascGroupType = GroupType::firstOrCreate([
+            'name' => 'National Student Council',
+        ]);
+
+        info('rogue:nassp-groups-import: Beginning import...');
+
+        /**
+         * Without specifying these column names, this command fails with error:
+         * "The header record must be an empty or a flat array with unique string value."
+         */
+        $records = $csv->getRecords(['LABEL NAME', 'NHS', 'NJHS', 'NASC', 'NASSP', 'ADDRESS1', 'ADDRESS2', 'CITY', 'STATE', 'POSTAL', 'COUNTRY', 'NCES']);
```

Comment:
Do we also need to specify the records that we're not using in this script?

Reply:
Yeah, without adding them -- the wrong values get used. I think this may be an issue with the source CSV column headers having null values (there's like 3 blank columns but sometimes weird stuff appears for some rows here ). The source file isn't ready for a production import yet, so I may have some time to clean this up - esp because it will need to be revisited in [per comment below](https://github.com/DoSomething/rogue/pull/1056#discussion_r445749156).